api/all/src/main/java/io/opentelemetry/api/metrics/MeterBuilder.java

```diff
@@ -28,6 +30,17 @@ public interface MeterBuilder {
    */
   MeterBuilder setInstrumentationVersion(String instrumentationScopeVersion);
 
+  /**
+   * Assigns instrumentation scope attributes for the resulting Meter.
+   *
+   * @param attributes The instrumentation scope attributes.
+   * @return this
+   * @since 1.18.0
+   */
+  default MeterBuilder setAttributes(Attributes attributes) {
```

Comment:
Can also make the argument the name should be `setInstrumentationAttributes`. The two setters currently don't align in this regard (`setSchemaUrl` and `setInstrumentationVersion`). I omitted "instrumentation" because it doesn't add any clarity - would have to be `setScopeAttributes` to improve clarify, which doesn't align with the existing naming pattern.

Reply:
`setStaticAttributes` ? `setFixedAttributes`? If we want to stay with `setAttributes` (and I'm not opposed necessarily), we should add more clarifying javadoc about what they will be used for.

Reply:
`setCommonAttributes` ?

Reply:
The setters on these builders are all defining properties for scope, but scopes are an implementation detail that don't appear the SDK makes them available to exporters via SpanData, MetricData, LogData. 

From the perspective of an API user, these properties just combine to define a unique tracer / meter / logger. I.e. if you specify a different name, version, schemaUrl, or attributes, you should [expect to get a different instance](https://github.com/open-telemetry/opentelemetry-specification/blame/main/specification/trace/api.md#L134-L137).

I think `setAttributes` is probably most correct, but all of these javadoc could use clarification that distinct name, version, schemaUrl, attribute combinations return unique instances.

Reply:
I've tried to improve the javadoc overall in LoggerBuilder, TracerBuilder, and MeterBuilder. Going to mark this as resolved but please reopen needed 🙂